### PR TITLE
Feature/tag reader zips

### PR DESCRIPTION
**Pull request by tznind**
## Proposed Changes

Support for zip files in source dicom file systems.  This is required for allowing architecture mirror between EPCC and HIC.  

Also removes the requirement for AccessionNumber to be encoded in the file system

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [x] Updated any relevant API documentation
- [x] Created or updated any tests if relevant
- [x] Accurately updated the [CHANGELOG](https://github.com/SMI/SmiServices/blob/master/CHANGELOG.md)
  - NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

### CODE REVIEW

File: src/common/Smi.Common/Messages/AccessionDirectoryMessage.cs [hunk 1]
```diff
@@ -14,7 +14,6 @@ public sealed class AccessionDirectoryMessage : IMessage
         /// <summary>
         /// NationalPACSAccessionNumber obtained from the end of the directory path.
         /// </summary>
-        [JsonProperty(Required = Required.Always)]
```
Comment (rkm) [inline src/common/Smi.Common/Messages/AccessionDirectoryMessage.cs:17]: We'll need some tests to check the behaviour when this is null.

File: src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs [hunk 2]
```diff
@@ -85,11 +86,13 @@ public void ReadTags(IMessageHeader header, AccessionDirectoryMessage message)
             if (!dirPath.StartsWith(_filesystemRoot, StringComparison.CurrentCultureIgnoreCase))
                 throw new ApplicationException("Directory " + dirPath + " is not below the given FileSystemRoot (" +
                                                _filesystemRoot + ")");
-
             long beginEnumerate = _stopwatch.ElapsedTicks;
             string[] dicomFilePaths = _fs.Directory.EnumerateFiles(dirPath, _searchPattern).ToArray();
+            string[] zipFilePaths =  _fs.Directory.EnumerateFiles(dirPath).Where(f=>Path.GetExtension(f) == ".zip" || Path.GetExtension(f) == ".tar").ToArray();
```
Comment (rkm) [inline src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs:91]: On Linux, I think we're more likely to see gzipped archives rather than bare tar files -- so *.tar.gz and *.tgz. We should be able to use GZipStream to handle this case though.

Comment (tznind) [reply src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs:91]: Do we want to move the IsZipFile into a helper class? so the definition is standard and if needed wrap the openread for different file types?

**Comment by tznind** [comment]
> I wonder if the class should be called "ArchiveDicomDirectoryFinder" rather than "Zip...", since we support multiple archive formats?

The format is `-f zips` for the CLI so would need to change there too presumably.  Also the full file system could be considered an archive of dicom files even uncompressed... and PACS has the word Archive in it's name so I think archive alone is probably a bit ambigious.

File: src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs (hunk 2 above)
Comment (rkm) [reply src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs:91]: I think that would make sense, yes. We're also trusting the file extension rather than checking the file contents before trying to read -- I assume the read will fail with a sensible error though if that happens.

Comment (tznind) [reply src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs]: Ok I have moved the extension evaluation to ZipHelper https://github.com/SMI/SmiServices/pull/327/commits/636aff7544c3643dad700e58eb1e75e3e85621c7 

If we need to enumerate different archive types and serve dicomdatasets out of them then I would rather move that class up to Rdmp.Dicom or DicomTypeTranslation since its quite a big piece of functionality and theres already archive handling stuff further up e.g. [ZipPool](https://github.com/HicServices/RdmpDicom/blob/52e10f9d323f63e2998c8794d0afa5587a0801de/Rdmp.Dicom/Extraction/FoDicomBased/ZipPool.cs) which handles random access into zip files.

**Comment by rkm** [comment]
Have added some tests for the accession number bit and fixed the associated code. LGTM now!
